services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -656,28 +693,41 @@ class AudioColumn(MediaColumn):
     transform_column = FloatColumn
 
     @staticmethod
-    def get_duration(example: dict[str, Any]) -> float:
+    def get_duration(example: Optional[dict[str, Any]]) -> Optional[float]:
         """Get audio durations"""
+        if example is None:
+            return None
         with io.BytesIO(example["bytes"]) as f:
             return librosa.get_duration(path=f)  # type: ignore   # expects PathLike but BytesIO also works
 
     @classmethod
-    def transform(cls, example: dict[str, Any]) -> float:
+    def transform(cls, example: Optional[dict[str, Any]]) -> Optional[float]:
         return cls.get_duration(example)
 
 
 class ImageColumn(MediaColumn):
     transform_column = IntColumn
 
     @staticmethod
-    def get_width(example: dict[str, Any]) -> int:
+    def get_width(example: Optional[dict[str, Any]]) -> Optional[int]:
         """Get image widths."""
+        if example is None:
+            return None
         with io.BytesIO(example["bytes"]) as f:
             image = Image.open(f)
             return image.size[0]
 
+    @staticmethod
+    def get_shape(example: Optional[dict[str, Any]]) -> Union[tuple[None, None], tuple[int, int]]:
```

Comment:
you might want to take into account the case where example is `bytes` ("bytes" only)